crates/uv-cli/src/lib.rs

```diff
@@ -449,21 +448,18 @@ pub struct HelpArgs {
 }
 
 #[derive(Args)]
-#[cfg(feature = "self-update")]
 pub struct SelfNamespace {
     #[command(subcommand)]
     pub command: SelfCommand,
 }
 
 #[derive(Subcommand)]
-#[cfg(feature = "self-update")]
 pub enum SelfCommand {
     /// Update uv.
     Update(SelfUpdateArgs),
 }
 
 #[derive(Args, Debug)]
-#[cfg(feature = "self-update")]
```

Comment:
Hi there, may I ask why these `#[cfg(feature = "self-update")]` changes were reverted?

Reply:
With those guards and `--no-default-features`, uv pretends the subcommands doesn't exist; with this change, it shows an error message hinting the user to upgrade using their package manager.